src/dictionary.c

```diff
@@ -272,6 +272,10 @@ static struct dictionary* new_dictionary_opts(SEXP x, struct dictionary_opts* op
   default: Rf_error("Internal error: Unimplemented type in `new_dictionary()`.");
   }
 
+  // Reprotect `d->protect` for the case of `init_dictionary_df()`,
```

Comment:
This comment is a bit too specific, it can just be "`init_` functions might allocate", if the protect is not self-explanatory.